datumaro/plugins/data_formats/segment_anything/exporter.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+
+import logging as log
+import os
+import os.path as osp
+from collections import defaultdict
+
+from pycocotools import mask as mask_utils
+
+from datumaro.components.annotation import AnnotationType
+from datumaro.components.errors import DatumaroError, MediaTypeError
+from datumaro.components.exporter import Exporter
+from datumaro.components.media import Image
+from datumaro.util import NOTSET, dump_json_file
+
+
+def replace(json_data, key, value_new):
+    value_origin = json_data[key]
+    if value_origin is NOTSET:
+        json_data[key] = value_new
+        return
+    if value_origin != value_new:
+        raise DatumaroError(f"The value for '{key}' is not same for item {value_new}")
+
+
+class SegmentAnythingExporter(Exporter):
+    DEFAULT_IMAGE_EXT = ".jpg"
+
+    def __init__(
+        self,
+        extractor,
+        save_dir,
+        **kwargs,
+    ):
+        super().__init__(extractor, save_dir, **kwargs)
+
+    def apply(self):
+        if self._extractor.media_type() and not issubclass(self._extractor.media_type(), Image):
+            raise MediaTypeError("Media type is not an image")
+
+        os.makedirs(self._save_dir, exist_ok=True)
+
+        subsets = self._extractor.subsets()
+        pbars = self._ctx.progress_reporter.split(len(subsets))
+
+        _image_ids = set()
+
+        image_id = 1
+        for pbar, (subset_name, subset) in zip(pbars, subsets.items()):
+            for item in pbar.iter(subset, desc=f"Exporting {subset_name}"):
+                try:
+                    # make sure file_name is flat
+                    file_name = self._make_image_filename(item).replace("/", "__")
+                    image_id = int(item.attributes.get("id", image_id))
```

Comment:
`item.id` is not likely guaranteed to be an integer because it is a string.

Reply:
Fixed in https://github.com/openvinotoolkit/datumaro/pull/1005/commits/207ac63b848e0398e4a366f1bda833dcfae54947